.github/workflows/test.yml

```diff
@@ -176,3 +176,8 @@ jobs:
 
       - name: Check Formatting
         uses: dprint/check@v2.2
+
+      - name: Check Cargo.lock is up-to-date
+        run: |
+          cargo build --workspace
```

Comment:
Not adding `--locked` param here because it will detect only that deps listed in Cargo.lock match what's needed to build the project but won't validate formatting issues such us newlines.

Without `--locked` param this should catch both bad version and formatting issues in `Cargo.lock`. Example of badly formatting `Cargo.lock` [here](https://github.com/google/mdbook-i18n-helpers/actions/runs/10270090217).

Reply:
This is going to take a while, can it be part of the `test` job and/or run `cargo check` rather than `cargo build`?

Reply:
`cargo build` takes 45 seconds without using cache and 14 seconds with cache enabled.
`cargo check` is 11 seconds with cache enabled. I am ok with either.

Replaced it with `cargo check` and moved it to `Test` job.